File: Python/Warmup-2/array123.py
"""
Given an array of ints, return True if the sequence of numbers 1, 2, 3 appears in the array somewhere.

array123([1, 1, 2, 3, 1]) → True
array123([1, 1, 2, 4, 1]) → False
array123([1, 1, 2, 1, 2, 3]) → True

@author unobatbayar
"""

def array123(nums):
  for x in range(0, len(nums)-2): # -2 because we're adding x + 2 and it will exceed
    if nums[x] == 1 and nums[x+1] == 2 and nums[x+2] == 3:
      return True
  return False

File: Java/Warmup-2/array123.java
/**
Given an array of ints, return true if the sequence of numbers 1, 2, 3 appears in the array somewhere.

array123([1, 1, 2, 3, 1]) → true
array123([1, 1, 2, 4, 1]) → false
array123([1, 1, 2, 1, 2, 3]) → true

@author unobatbayar
*/

public boolean array123(int[] nums) {
  for (int i = 0; i <nums.length-2; i++){
    if (nums[i] == 1 && nums[i+1] == 2 && nums[i+2] == 3){
      return true;
    }
  }
  return false;
}
